### ISSUE TITLE
error in garden search

### ISSUE THREAD

**Issue by AlessandroMedici** [kivy/kivy#4930]
Lanuncing: garden search
Traceback:
AlessandroMedici/garden_search_traceback

I think it's just a simple omitted line:

before:
        p = subparsers.add_parser('search',
                help='Search garden package on github')
        p.add_argument('pattern', nargs='?', default='',
                help='Word to search in the package name (optional)')
        p.set_defaults(func=self.cmd_search)
and I get the attributeError.

simple pasting p.add_add_argument("--kivy"... :
        p = subparsers.add_parser('search',
                help='Search garden package on github')
        p.add_argument('--kivy', action='store_true',help='Use the local app directory (./libs/garden)')
        p.add_argument('pattern', nargs='?', default='',
                help='Word to search in the package name (optional)')
        p.set_defaults(func=self.cmd_search)
And all is right.

But I don't know if this attributeError is a voluntary error,   perhaps for trapping it in calling progamm.

**Comment by dessant** [comment]
It's hard to follow your post, could you share the full traceback, together with the command you used? Garden version? How did you install it?

**Comment by AlessandroMedici** [comment]
Ok, sorry but no time :-(

1) enabled in ...//.kivy/config.ini:
[kivy]
log_level = debug

2) running:
garden search

-> no logs at all, obviosly: no error in kivy, only in garden.

Python 3.5.3rc1 (default, Jan  3 2017, 04:40:57) 
[GCC 6.3.0 20161229] on linux Debian-Sid

python3-kivy package: v.1.9.1-2 stable

Traceback:

alex@del:/usr/local/bin$ garden search
anchoredscrollview - Anchored ScrollView Kivy widget
androidtabs - No description
androidvlc - No description
animlabel - An animated label implementation, allowing for custom transform functions
animlabel_bezier - an extension of the animlabel tool, to make a label follow a bezier path
bar - Bar widget represents the 0-100 range percentage value as a coloured rectangle. It is known as used in statistics graphs.
cefpython - CEFPython Kivy widget
circulardatetimepicker - Android-like date & time picker for Kivy
circularlayout - CircularLayout for Kivy
collider - Collides a point with a space defined by a curve.
contextmenu - No description
datetimepicker - roulette based datetime selector like in iOS and android.
ddd - A 3d view widget for kivy
desktopvideoplayer - No description
docbrowser - Browser for view documentation of Python modules
facelock - It is a flower made for kivy-garden
filebrowser - An advanced file browser.
filechooserthumbview - FileChooserThumbView for Kivy
gauge - Kivy Widget Gauge
geartick - A gear based Ticker for use in rotating sliders 
graph - Displays plots on a graph.
hotkeybehavior - Global HotKey behavior for a Kivy root widget
iconfonts - Simple helper functions to make easier to use icon fonts in Labels and derived widgets
inkcanvas - Gesture recognition
knob - Knob widget for Kivy.
magnet - A magnet widget for kivy garden
mapview - Mapview is a Kivy widget for displaying interactive maps.
matplotlib - Matplotlib backends using kivy
modernmenu - a stylized menu system for kivy
Traceback (most recent call last):
  File "/usr/local/bin/garden", line 229, in <module>
    GardenTool().main(sys.argv[1:])
  File "/usr/local/bin/garden", line 96, in main
    if self.options.kivy and garden_kivy_dir is None:
AttributeError: 'Namespace' object has no attribute 'kivy'

##########
# Simple adding this line in "garden" file at line 55 solved problem:
p.add_argument('--kivy', action='store_true',help='Use the local app directory (./libs/garden)')
##########

**Comment by AlessandroMedici** [comment]
Ooops: sorry for too big:

Simple adding this line in "garden" file at line 55 solved problem:

Not in my mind.

And many thanks for you big work!!!

**Comment by mivade** [comment]
I see the same thing after installing via conda:

```
$ garden search
[INFO   ] [Logger      ] Record log in ~/.kivy/logs/kivy_21-08-17_4.txt
[INFO   ] [Kivy        ] v2.0.0
[INFO   ] [Kivy        ] Installed at "~/miniconda3/envs/kivy/lib/python3.9/site-packages/kivy/__init__.py"
[INFO   ] [Python      ] v3.9.6 | packaged by conda-forge | (default, Jul 11 2021, 03:36:15)
[Clang 11.1.0 ]
[INFO   ] [Python      ] Interpreter at "~/miniconda3/envs/kivy/bin/python3.9"
anchoredscrollview - Anchored ScrollView Kivy widget
androidtabs - No description
androidvlc - No description
animlabel - An animated label implementation, allowing for custom transform functions
animlabel_bezier - an extension of the animlabel tool, to make a label follow a bezier path
bar - Bar widget represents the 0-100 range percentage value as a coloured rectangle. It is known as used in statistics graphs.
cefpython - CEFPython Kivy widget
circulardatetimepicker - Android-like date & time picker for Kivy
circularlayout - CircularLayout for Kivy
collider - Collides a point with a space defined by a curve. DEPRECATED for https://github.com/kivy-garden/collider
contactform - Kivy garden flower providing widget for sending e-mail from form on app
contextmenu - Migrated to https://github.com/kivy-garden/contextmenu
datetimepicker - roulette based datetime selector like in iOS and android.
ddd - A 3d view widget for kivy
desktopvideoplayer - No description
docbrowser - Browser for view documentation of Python modules
facelock - It is a flower made for kivy-garden
filebrowser - An advanced file browser. DEPRECATED for https://github.com/kivy-garden/filebrowser
filechooserthumbview - FileChooserThumbView for Kivy
gauge - Kivy Widget Gauge
geartick - A gear based Ticker for use in rotating sliders
 Traceback (most recent call last):
   File "~/miniconda3/envs/kivy/bin/garden", line 223, in <module>
     GardenTool().main(sys.argv[1:])
   File "~/miniconda3/envs/kivy/bin/garden", line 90, in main
     if self.options.kivy and garden_kivy_dir is None:
 AttributeError: 'Namespace' object has no attribute 'kivy'
```
